src/app/services/user.ts

```diff
@@ -3,66 +3,70 @@ import { LoggedInUser, PotentialUser, School } from "../../IsaacAppTypes";
 import { UserRole } from "../../IsaacApiTypes";
 import { Immutable } from "immer";
 
+type UserType = {
```

Comment:
I think this name needs refactoring: you don't need to include the word `Type` in it, and this is more than a user, this is a combination of a user and their logged in status.

Reply:
What about something like `UserSession` or `LoginStatus`? They're not great, so feel free to use something different.